assets/js/googlesitekit/datastore/site/urls.test.js

```diff
@@ -77,6 +77,153 @@ describe( 'core/site urls', () => {
+			} );
+		} );
+
+		describe( 'getErrorTroubleshootingLinkURL', () => {
+			const testError = {
+				data: {
+					reason: 'Data Error',
+				},
+				selectorData: {
+					args: [
+						{
+							dimensions: [ 'ga:date' ],
+							metrics: [ { expression: 'ga:users' } ],
+							startDate: '2020-08-11',
+							endDate: '2020-09-07',
+						},
+					],
+					name: 'getReport',
+					storeName: 'MODULES_ANALYTICS',
+				},
+			};
+
+			const testErrorWithMessage = {
+				...testError,
+				message: 'Test error message',
+			};
+			const testErrorWithCode = { ...testError, code: 'test-error-code' };
+			const testErrorWithID = { ...testError, id: 'test-error-id' };
+
+			it( 'should throw an error if no error is given', () => {
+				expect( () =>
+					registry
+						.select( CORE_SITE )
+						.getErrorTroubleshootingLinkURL()
+				).toThrow( 'An error is required.' );
+			} );
+
+			it( 'should return the support link with the error message if no code or id is given', () => {
+				const baseURL = registry
+					.select( CORE_SITE )
+					.getProxySupportLinkURL();
+
+				const url = registry
+					.select( CORE_SITE )
+					.getErrorTroubleshootingLinkURL( testErrorWithMessage );
+
+				expect( url ).toBe(
+					`${ baseURL }?error=${ encodeURIComponent(
+						testErrorWithMessage.message
+					) }`
+				);
+			} );
+
+			it( 'should return the support link with the error code if the error code is given', () => {
+				const baseURL = registry
+					.select( CORE_SITE )
+					.getProxySupportLinkURL();
+
+				const url = registry
+					.select( CORE_SITE )
+					.getErrorTroubleshootingLinkURL( testErrorWithCode );
+
+				expect( url ).toBe(
+					`${ baseURL }?error_id=${ encodeURIComponent(
+						testErrorWithCode.code
+					) }`
+				);
+			} );
+
+			it( 'should return the support link with the error id if the error id is given', () => {
+				const baseURL = registry
+					.select( CORE_SITE )
+					.getProxySupportLinkURL();
+
+				const url = registry
+					.select( CORE_SITE )
+					.getErrorTroubleshootingLinkURL( testErrorWithID );
+
+				expect( url ).toBe(
+					`${ baseURL }?error_id=${ encodeURIComponent(
+						testErrorWithID.id
+					) }`
+				);
+			} );
+
+			it( 'it should default to the error message if the error code is a numeric value', () => {
+				const testErrorWithNumericCode = {
+					...testErrorWithMessage,
+					code: 123,
+				};
+
+				const baseURL = registry
+					.select( CORE_SITE )
+					.getProxySupportLinkURL();
+
+				const url = registry
+					.select( CORE_SITE )
+					.getErrorTroubleshootingLinkURL( testErrorWithNumericCode );
+
+				expect( url ).toBe(
+					`${ baseURL }?error=${ encodeURIComponent(
+						testErrorWithNumericCode.message
+					) }`
+				);
+			} );
+
+			it( 'it should default to the error message if the error id is a numeric value', () => {
+				const testErrorWithNumericID = {
+					...testErrorWithMessage,
+					id: 123,
```

Comment:
The ACs for both this and the `code` property state that they should be a non-numeric string… the implication to me being that a string like `'403'` should _not_ be used as an error ID/code. The ACs explicitly spell this out:

> in other words it should be textual slug, not a number

So we should add a test similar to this one, but with an `id` of `'123'`. Currently it would fail, so some changes to the logic in `getErrorTroubleshootingLinkURL` would be needed. (Same goes for `error.code`.)

Reply:
Thanks for catching this, @tofumatt. My apologies for not catching this in my CR.

Reply:
Hey! I've added a new `isNumeric` util that adds a stricter check, see the new tests accompanying it.